core/src/value/deserialize.rs

```diff
@@ -29,55 +50,286 @@ pub enum Error {
+    Second(u16, u8, u8, u8, u8, u8),
+    Fraction(u16, u8, u8, u8, u8, u8, u32),
+}
+
+#[derive(Debug, PartialEq)]
+pub struct PartialDateTime {
+    components: DateTimePrecision,
+    // TODO probably not an option, will see ..
+    offset: Option<FixedOffset>,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum TimePrecision {
+    Hour(u8),
+    Minute(u8, u8),
+    Second(u8, u8, u8),
+    Fraction(u8, u8, u8, u32),
+}
+
+#[derive(Debug, PartialEq)]
+pub struct PartialTime {
+    components: TimePrecision,
+}
+
+/**
+ * This trait is implemented by partial precision
+ * Date, Time and DateTime structures.
+ * Trait method returns the last fully precise component of the value.
+ */
+pub trait Precision {
+    fn precision(&self) -> DateComponent;
+}
+
+impl Precision for PartialDate {
+    fn precision(&self) -> DateComponent {
+        match self.components {
+            DatePrecision::Year(..) => DateComponent::Year,
+            DatePrecision::Month(..) => DateComponent::Month,
+            DatePrecision::Day(..) => DateComponent::Day,
+        }
+    }
+}
+
+/**
+ * This trait is implemented by partial precision
+ * Date, Time and DateTime structures.
+ */
+pub trait Imprecision<U>
+where
+    Self: Precision,
+    U: PartialEq,
+{
+    /**
+     * Returns the earliest possible value from a partial precision structure.
+     * So missing components default to 1 (days, months) or 0 (hours, minutes, ...)
+     * If structures contain invalid combination of date/time components, it fails.
+     */
+    fn earliest(&self) -> Result<U>;
+
+    /**
+     * Returns the latest possible value from a partial precision structure.
+     * If structures contain invalid combination of date/time components, it fails.
+     */
+    fn latest(&self) -> Result<U>;
+
+    /**
+     * Returns a tuple of the earliest and latest possible value from a partial precision structure.
+     * 
+     */
+    fn to_range(&self) -> Result<(Option<U>, Option<U>)> {
+        if self.is_precise() {
+            ZeroRange.fail()
+        } else {
+            Ok((self.earliest().ok(), self.latest().ok()))
+        }
+    }
+
+    /**
+     * Returns true, if partial precision structure has maximum possible accuracy.
+     * For fraction of a second, ... TODO decide what to do, as RUST Chrono is more
+     * precise than 6 digits of DICOM standard ...
+     */
+    fn is_precise(&self) -> bool {
+        let e = self.earliest();
+        let l = self.latest();
+
+        e.is_ok() && l.is_ok() && e.ok() == l.ok()
+    }
+}
+
+impl Imprecision<NaiveDate> for PartialDate {
+    fn earliest(&self) -> Result<NaiveDate> {
+        let (y, m, d) = match self.components {
+            DatePrecision::Year(y) => (y, 1, 1),
+            DatePrecision::Month(y, m) => (y, m, 1),
+            DatePrecision::Day(y, m, d) => (y, m, d),
+        };
+        let date: Result<_> =
+            NaiveDate::from_ymd_opt(y as i32, m as u32, d as u32).context(InvalidDate);
+        Ok(date?)
```

Comment:
This should work. A similar pattern can be employed in the other methods.
```suggestion
        NaiveDate::from_ymd_opt(y as i32, m as u32, d as u32).context(InvalidDate)
```